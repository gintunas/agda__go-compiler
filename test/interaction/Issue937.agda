-- 2013-11-xx Andreas
-- Previous clauses did not reduce in later clauses under
-- a module telescope.

{-# OPTIONS --copatterns #-}
-- {-# OPTIONS -v interaction.give:20 -v tc.cc:60 -v reify.clause:60 -v tc.section.check:10 -v tc:90 #-}
-- {-# OPTIONS -v tc.lhs:20 #-}
-- {-# OPTIONS -v tc.cover:20 #-}

module Issue937 where

data Nat : Set where
  zero : Nat
  suc : Nat → Nat
{-# BUILTIN NATURAL Nat #-}

infixr 4 _,_

record Σ (A : Set) (B : A → Set) : Set where
  constructor _,_
  field
    proj₁ : A
    proj₂ : B proj₁

open Σ public

data _≤_ : Nat → Nat → Set  where
  z≤n : ∀ {n}                 → zero  ≤ n
  s≤s : ∀ {m n} (m≤n : m ≤ n) → suc m ≤ suc n

_<_ : Nat → Nat → Set
m < n = suc m ≤ n

ex : Σ Nat (λ n → zero < n)
proj₁ ex = suc zero
proj₂ ex = s≤s z≤n -- works

ex' : Σ Nat (λ n → zero < n)
proj₁ ex' = suc zero
proj₂ ex' = {! s≤s z≤n !}

module _ (A : Set) where

  ex'' : Σ Nat (λ n → zero < n)
  proj₁ ex'' = suc zero
  proj₂ ex'' = s≤s z≤n -- works

  ex''' : Σ Nat (λ n → zero < n)
  proj₁ ex''' = suc zero
  proj₂ ex''' = {! s≤s z≤n !}

-- The normalized goals should be printed as 1 ≤ 1
