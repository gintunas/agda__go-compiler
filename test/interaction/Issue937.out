(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : zero < proj₁ ex' ?1 : zero < proj₁ ex''' " nil)
((last . 1) . (agda2-goals-action '(0 1)))
(agda2-status-action "")
(agda2-info-action "*Current Goal*" "1 ≤ 1" nil)
(agda2-status-action "")
(agda2-info-action "*Current Goal*" "1 ≤ 1" nil)
